config/v1/types_authentication.go

```diff
@@ -565,27 +601,44 @@ type PrefixedClaimMapping struct {
 	Prefix string `json:"prefix"`
 }
 
+// TokenValidationRuleType defines the type of token validation rule.
+//
+// +kubebuilder:validation:Enum=RequiredClaim;Expression
 type TokenValidationRuleType string
 
 const (
-	TokenValidationRuleTypeRequiredClaim = "RequiredClaim"
+	TokenValidationRuleRequiredClaim TokenValidationRuleType = "RequiredClaim"
```

Comment:
I believe setting the type for this when it wasn't set before changes how Go interprets the value. This is a breaking change to anyone importing openshift/api and using this value.

I would generally avoid making a change like this unless you can enumerate all the places that may be broken by this change.

Reply:
For reference, projects importing and using this value will likely run into build error like this anywhere they are using `TokenValidationRuleRequireClaim` as a string value:
```
cannot use TokenValidationRuleRequiredClaim (constant "RequiredClaim" of string type TokenValidationRuleType) as string value in assignment
```